libs/ui/date-picker/helm/src/lib/hlm-date-picker.component.ts

```diff
@@ -0,0 +1,135 @@
+import { HlmCalendarComponent } from '@spartan-ng/ui-calendar-helm';
+import { HlmIconDirective } from '@spartan-ng/ui-icon-helm';
+import { HlmPopoverContentDirective } from '@spartan-ng/ui-popover-helm';
+import type { ClassValue } from 'clsx';
+
+export const HLM_DATE_PICKER_VALUE_ACCESSOR = {
+	provide: NG_VALUE_ACCESSOR,
+	useExisting: forwardRef(() => HlmDatePickerComponent),
+	multi: true,
+};
+
+// TODO make date picker work with forms
+@Component({
+	selector: 'hlm-date-picker',
+	standalone: true,
+	imports: [
+		NgIcon,
+		HlmIconDirective,
+		BrnPopoverComponent,
+		BrnPopoverTriggerDirective,
+		BrnPopoverContentDirective,
+		HlmPopoverContentDirective,
+		HlmCalendarComponent,
+	],
+	providers: [HLM_DATE_PICKER_VALUE_ACCESSOR, provideIcons({ lucideCalendar })],
+	template: `
+		<brn-popover sideOffset="5" closeDelay="100">
+			<button type="button" [class]="_computedClass()" [disabled]="state().disabled()" brnPopoverTrigger>
+				<ng-icon hlm size="sm" name="lucideCalendar" />
+
+				@if (formattedDate(); as formattedDate) {
+					{{ formattedDate }}
+				} @else {
+					<ng-content />
+				}
+			</button>
+
+			<div hlmPopoverContent class="w-auto p-0" *brnPopoverContent="let ctx">
+				<hlm-calendar
+					[date]="date()"
+					[min]="min()"
+					[max]="max()"
+					[disabled]="state().disabled()"
+					(dateChange)="_handleChange($event)"
+				/>
+			</div>
+		</brn-popover>
+	`,
+	host: {
+		class: 'block',
+	},
+})
+export class HlmDatePickerComponent<T> {
+	public readonly userClass = input<ClassValue>('', { alias: 'class' });
+	protected readonly _computedClass = computed(() =>
+		hlm(
+			'inline-flex items-center gap-2 whitespace-nowrap rounded-md text-sm ring-offset-background transition-colors border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 w-[280px] justify-start text-left font-normal',
+			'focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2',
+			'disabled:pointer-events-none disabled:opacity-50',
+			'[&_ng-icon]:pointer-events-none [&_ng-icon]:shrink-0',
+			!this.date() ? 'text-muted-foreground' : '',
+			this.userClass(),
+		),
+	);
+
+	/** The minimum date that can be selected.*/
+	public readonly min = input<T>();
+
+	/* * The maximum date that can be selected. */
+	public readonly max = input<T>();
+
+	/** Determine if the date picker is disabled. */
+	public readonly disabled = input<boolean, BooleanInput>(false, {
+		transform: booleanAttribute,
+	});
+
+	/** The selected value. */
+	public readonly date = model<T>();
+
+	protected readonly state = computed(() => ({
+		disabled: signal(this.disabled()),
+	}));
+
+	public readonly dateFormat = input<(date: T) => string>((date) =>
+		date instanceof Date ? date.toDateString() : `${date}`,
+	);
+
+	protected readonly formattedDate = computed(() => {
+		const date = this.date();
+
+		if (!date) {
+			return undefined;
+		}
+
+		return this.dateFormat()(date);
+	});
```

Comment:
@ashley-hunter I want to allow the date to be formatted easily. Atm a format function can be passed to `dateFormat` input, which has a default. Do you think this could be part of the `BrnDateAdapter` or perhaps its own `HlmFormatAdapter`, thus a default could be provided for all date-picker through out the applicaiton.

Reply:
Sure, yes, certainly could make sense to allow a default to be set. The pattern we have been using for this in other components is a provide config approach, which can be seen here. I think a similar pattern could be used for this too:

https://github.com/spartan-ng/spartan/blob/main/libs/ui/icon/helm/src/lib/hlm-icon.token.ts

Reply:
Thanks, thats a good idea. I added the global config also as an example and how to override it on a component basis.